lean4-infoview/src/infoview/tooltips.tsx

```diff
@@ -71,57 +70,43 @@ export function Tooltip(props_: TooltipProps) {
   return ReactDOM.createPortal(floating, document.body)
 }
 
-export const WithFadingTooltip =
+export const WithToggleableTooltip =
   forwardAndUseStateRef<HTMLSpanElement, React.HTMLProps<HTMLSpanElement> & {
     isTooltipShown: boolean
     hideTooltip: () => void
     tooltipChildren: React.ReactNode
   }>((props_, ref, setRef) => {
   const { isTooltipShown, hideTooltip, tooltipChildren, ...props } = props_
 
-  const fadeOutStyles: { [status in TransitionStatus]: any } = {
-    'entering': {
-      opacity: 1
-    },
-    'entered': {
-      opacity: 1
-    },
-    'exiting': {
-      opacity: 0,
-      transition: 'opacity 1.5s ease-out',
-    },
-    'exited': {
-      display: 'none',
-      opacity: 0
-    },
-    'unmounted': {
-      opacity: 0
-    }
-  }
-
-  // Workaround for warning generated by react-transition-group.
-  // See https://github.com/reactjs/react-transition-group/issues/668#issuecomment-791003824.
-  const nodeRef = React.useRef(null)
+  // Since we do not want to hide the tooltip if the user is trying to select text in it,
+  // we need both the "click outside" and "click inside" handlers here because they
+  // play nicer with existing selections than a global click handler.
+  // With a single global click handler, any selection anywhere in the InfoView could block
+  // the tooltip from being hidden. This is especially annoying because right-clicking any
+  // element also selects it.
+  // With both inside and outside click handlers, the outside click handler can simply disregard
+  // selections, whereas React ensures that only a selection in the tooltip itself can block
+  // the inside click handler from hiding the tooltip, since the outer selection is removed
+  // before the inside click handler fires.
+  const [logicalSpanElt, logicalDomStorage] = useLogicalDomObserver({current: ref})
+  const onClickOutside = React.useCallback(() => {
+    hideTooltip()
+  }, [hideTooltip])
+  useOnClickOutside(logicalSpanElt, onClickOutside)
 
-  return (<span ref={setRef} {...props}>
-    <Transition
-      nodeRef={nodeRef}
-      in={isTooltipShown}
-      timeout={1500}
-      onEntered={() => hideTooltip()}
-    >
-    {state =>
-      <span ref={nodeRef}>
-        <Tooltip
-          reference={ref}
-          style={fadeOutStyles[state]}
-        >
-          {tooltipChildren}
-        </Tooltip>
-      </span>}
-    </Transition>
-    {props.children}
-  </span>)
+  return <LogicalDomContext.Provider value={logicalDomStorage}>
+      <span
+        ref={setRef}
+        onClick={() => { if (!window.getSelection()?.toString()) hideTooltip() } }
```

Comment:
This has the effect that a selection within the error tooltip keeps it open, but a click there hides it. If that's intended, great; otherwise you can use `isWithinHoverable` as in `WithTooltipOnHover`.

Reply:
Yes, the intended behavior is that you can still copy and paste the error message :-)